web/src/components/redactors/EditRedactor.tsx

```diff
@@ -0,0 +1,433 @@
+import React, { useEffect, useReducer } from "react";
+import { Link, useHistory, useRouteMatch } from "react-router-dom";
+import { KotsPageTitle } from "@components/Head";
+import AceEditor, { Marker } from "react-ace";
+import "brace/mode/text";
+import "brace/mode/yaml";
+import "brace/theme/chrome";
+
+import Loader from "../shared/Loader";
+import { Utilities } from "../../utilities/utilities";
+
+import "../../scss/components/redactors/EditRedactor.scss";
+import Icon from "../Icon";
+
+import { KotsParams } from "@types";
+import { useSelectedApp } from "@features/App";
+
+type State = {
+  activeMarkers?: Marker[];
+  createConfirm: boolean;
+  createErrMsg: string;
+  creatingRedactor: boolean;
+  editConfirm: boolean;
+  editingErrMsg: string;
+  editingRedactor: boolean;
+  isLoadingRedactor: boolean;
+  redactorEnabled: boolean;
+  redactorErrMsg: string;
+  redactorName: string;
+  redactorYaml: string;
+};
+
+// TODO: upgrade AceEditor so we can use the type definitions
+let aceEditor: AceEditor | null;
+
+const EditRedactor = () => {
+  const [state, setState] = useReducer(
+    (currentState: State, newState: Partial<State>) => ({
+      ...currentState,
+      ...newState,
+    }),
+    {
+      createConfirm: false,
+      createErrMsg: "",
+      creatingRedactor: false,
+      editConfirm: false,
+      editingErrMsg: "",
+      editingRedactor: false,
+      isLoadingRedactor: false,
+      redactorEnabled: false,
+      redactorErrMsg: "",
+      redactorName: "",
+      redactorYaml: "",
+    }
+  );
+
+  const history = useHistory();
+  const match = useRouteMatch<KotsParams>();
+  const slug = useSelectedApp()?.selectedApp?.slug || "";
```

Comment:
I've been thinking about how to standardize our hooks interfaces. i def like the idea of returning `react-query`'s api but `selectedApp` is fully custom hook and it might be better to follow the react-router approach and just return `selectedApp` and not `{ selectedApp }` 

Also thinking that on the `react-query` returns we could add something like `dataFlattened` and just add the stuff that components use from the responses in a flattened data structure instead of spreading the api structure all over the app.

Reply:
curious how that would look like! I down to give it a try